src/app/features/auth/pages/registration-page/registration-page.component.ts

```diff
@@ -1,11 +1,78 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { AsyncPipe, NgClass, NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, DestroyRef, inject } from '@angular/core';
+import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { TuiButton, TuiError } from '@taiga-ui/core';
+import { TuiInputModule, TuiInputPasswordModule } from '@taiga-ui/legacy';
+import { TUI_VALIDATION_ERRORS, TuiFieldErrorPipe } from '@taiga-ui/kit';
+import { builInErrors } from '@shared/constants/built-in-errors.constant';
+import { matchPasswordsValidator } from '@features/auth/validators/password.validator';
+import { RouterLink } from '@angular/router';
+import { emailValidator } from '@features/auth/validators/email.validator';
+import { AuthService } from '@features/auth/services/auth/auth.service';
 
 @Component({
   selector: 'dd-registration-page',
   standalone: true,
-  imports: [],
+  imports: [
+    AsyncPipe,
+    ReactiveFormsModule,
+    TuiInputModule,
+    TuiInputPasswordModule,
+    TuiButton,
+    TuiError,
+    TuiFieldErrorPipe,
+    NgIf,
+    NgClass,
+    RouterLink,
+  ],
   templateUrl: './registration-page.component.html',
   styleUrl: './registration-page.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: TUI_VALIDATION_ERRORS,
+      useValue: builInErrors,
+    },
+  ],
 })
-export class RegistrationPageComponent {}
+export class RegistrationPageComponent {
+  private fb: FormBuilder = inject(FormBuilder);
+
+  private authService: AuthService = inject(AuthService);
+
+  private readonly destroy = inject(DestroyRef);
+
+  private isSubmitted = false;
+
+  registrationForm: FormGroup = this.fb.group({
+    email: this.fb.control(''),
+    password: this.fb.control(''),
+    confirmPassword: this.fb.control(''),
+    name: this.fb.control(''),
+  });
+
+  onSubmit() {
+    const emailControl = this.registrationForm.get('email');
+    const passwordControl = this.registrationForm.get('password');
+
+    this.isSubmitted = true;
+    emailControl?.setValidators([Validators.required, emailValidator()]);
+    emailControl?.updateValueAndValidity();
+    passwordControl?.setValidators([Validators.required, Validators.minLength(8)]);
+    passwordControl?.updateValueAndValidity();
+    this.registrationForm.setValidators([matchPasswordsValidator('password', 'confirmPassword')]);
+
+    this.registrationForm.markAllAsTouched();
```

Comment:
I suggest to make `isSubmitted` signal and put this logic into effect, for example. Also, check if value is already true to not execute this logic on every submit.